internal/bootstrapping/agent.go

```diff
@@ -357,27 +357,29 @@ func (a *Agent) HandleResponse(inboxMsg *message.Message) ([]*message.Message, e
 	return nil, err
 }
 
-func (a *Agent) handleResponseMessage(msg *message.Message) (string, *MessageError, error) {
+func (a *Agent) handleResponseMessage(msg *message.Message) (string, *MessageError, bool, error) {
 	var rspMsg *protocol.Envelope
 	if err := json.Unmarshal(msg.Payload, &rspMsg); err != nil {
 		a.Logger.Error("Unexpected message format", err, watermill.LogFields{
 			"uuid": msg.UUID,
 		})
-		return "", NewMessageParameterInvalidError(), err
+		return "", NewMessageParameterInvalidError(), false, err
 	}
 
 	correlationID := rspMsg.Headers.CorrelationID()
+	respReq := rspMsg.Headers.IsResponseRequired()
+
 	var replyToID string
-	// if rspMsg.Headers.IsResponseRequired() { // Uncomment when ditto header default value impl is fixed
-	if len(correlationID) > 0 {
+
+	if respReq && len(correlationID) > 0 {
```

Comment:
If the replyToID is removed, we don't need this logic. Even more, it doesn't seem to be redundant.